public/plugin.tsx

```diff
@@ -307,13 +307,15 @@ export class ObservabilityPlugin
       mount: appMountWithStartPage('metrics'),
     });
 
-    core.application.register({
-      id: observabilityApplicationsID,
-      title: observabilityApplicationsTitle,
-      category: OBSERVABILITY_APP_CATEGORIES.observability,
-      order: observabilityApplicationsPluginOrder,
-      mount: appMountWithStartPage('applications'),
-    });
+    if (!setupDeps.dataSource) {
+      core.application.register({
+        id: observabilityApplicationsID,
+        title: observabilityApplicationsTitle,
+        category: OBSERVABILITY_APP_CATEGORIES.observability,
+        order: observabilityApplicationsPluginOrder,
+        mount: appMountWithStartPage('applications'),
+      });
+    }
```

Comment:
What happens when we have enabled the new navigation and MDS together? Do have to modify `plugin_nav` file to not register applications there where MDS is enabled?

Reply:
It does not break anything as the core navigation group just fails to find observabilityApplicationsID and doesn't register anything to the left side. It would be cleaner to handle this condition though.

Reply:
> just fails to find observabilityApplicationsID

@TackAdam @sumukhswamy  Is this leading to some error on browser or Dashboards app? If so then let's handle that in this PR

Reply:
no error seen